### PR TITLE
[explorev2] using label in 'Visualization Type' Select instead of key

### PR DESCRIPTION
**Pull request by mistercrunch**
* moved url related logic upstream in the caller
* moved option creation logic from render method to the constructor as
  it only needs to be executed once

### CODE REVIEW

File: superset/assets/javascripts/explorev2/components/SelectField.jsx [hunk 1]
```diff
@@ -26,6 +26,40 @@ const defaultProps = {
 };
 
 export default class SelectField extends React.Component {
+  constructor(props) {
+    super(props);
+    const options = this.props.choices.map((c) => {
```
Comment (ascott) [inline superset/assets/javascripts/explorev2/components/SelectField.jsx]: what do you think of pulling  L31-58 into another function called `getOptions` to keep that logic out of the constructor?

**Comment by ascott** [comment]
small stylistic comment, otherwise LGTM.

File: superset/assets/javascripts/explorev2/components/SelectField.jsx (hunk 1 above)
Comment (mistercrunch) [inline superset/assets/javascripts/explorev2/components/SelectField.jsx]: sounds very reasonable and future proof for if/when we'll need to refresh options

**Comment by ascott** [comment]
🎉

**Pull request merged**